opendatahub-io/odh-dashboard
# Support storage size unit and show error message correctly

Closes #1480 in commit 1
Closes #1565 in commit 2

## Description


1. Refactoring the storage section in the storage modal and the workbench spawner page (they reuse the same component), adding unit support (Gi and Mi) for the storage size, and only allowing the users to input integers.
2. Add a dry run for the storage modal network calls, it can now throw errors as expected (Address the comment in #1679)

On the spawner page:
<img width="1724" alt="Screenshot 2023-08-29 at 9 39 19 AM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/88291b33-9cba-4eb9-8520-ed6e3df25a85">

In storage modal:
<img width="1724" alt="Screenshot 2023-08-29 at 9 40 52 AM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/9da7c3a2-a00b-4aff-9284-081321544391">

## How Has This Been Tested?



1. Go to the data science project and create a workbench with the storage
2. Try to edit the storage, give it a value smaller than the current size
3. You should be able to see the error alert in the modal without closing the modal
4. Try to change the size with different units, for example, if the original size is 1 Gi, try to change it to 2000 Mi
5. You should be able to make the change, and when you try to edit the storage, the size should be set correctly

## Test Impact


N/A, mostly network call stuff and reused components

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
